common/v2/services/Store/StoreProvider.tsx

```diff
@@ -51,6 +54,15 @@ export const StoreProvider = ({ children }: { children: React.ReactNode }) => {
       getAccountsAssetsBalances(accounts).then((accountsWithBalances: StoreAccount[]) => {
         // Avoid the state change if the balances are identical.
         if (isArrayEqual(accounts, accountsWithBalances)) return;
+        const accountsToCheck = accounts
+          .filter(account => account.networkId === 'Ethereum')
+          .filter(account => account.wallet !== WalletId.VIEW_ONLY);
+        getAccountsUnlockVIPStatus(accountsToCheck).then((unlockAccounts: string[]) => {
+          if (!unlockAccounts || unlockAccounts.length === 0) {
+            return;
+          }
+          setIsUnlockVerified(true);
+        });
```

Comment:
Avoid nested `then`'s. It's the equivalent of [callback hell](http://callbackhell.com/)
Prefer Promise chaining:
```
   getAccountsAssetsBalancess(accounts)
       .then(accountsWithBalances => {
           if (isArrayEqual(accounts, accountsWithBalances)) return;
           setAccounts(accountsWithBalances);
           return accountsToCheck = accounts
              .filter(account => account.networkId === 'Ethereum')
              .filter(account => account.wallet !== WalletId.VIEW_ONLY);
       })
       .then(getAccountsUnlockVIPStatus)
       .then(setIsUnlockVerified)
```
or use async/await